src/checkers/inference/VariableAnnotator.java

```diff
@@ -1172,8 +1172,12 @@ private void shallowAnnotateArray(AnnotatedTypeMirror type, Tree tree, int level
             addPrimaryVariable(type, tree);
         } else {
             TreePath pathToTopLevelTree = inferenceTypeFactory.getPath(topLevelTree);
-            ASTRecord astRecord = ASTPathUtil.getASTRecordForPath(inferenceTypeFactory, pathToTopLevelTree).newArrayLevel(level);
-            createEquivalentSlotConstraints(type, tree, new AstPathLocation(astRecord));
+            AstPathLocation location = null;
+            if (pathToTopLevelTree != null) {
+                ASTRecord astRecord = ASTPathUtil.getASTRecordForPath(inferenceTypeFactory, pathToTopLevelTree).newArrayLevel(level);
```

Comment:
Can you add a comment explaining when this branch is taken?

Reply:
As discussed, @bitterfox will add a short comment that for the artificial varargs trees we don't need an ASTPath.
After that change we can merge this.